samples/sqltransport/store-and-forward/sample.md

```diff
@@ -0,0 +1,101 @@
+ 1. Start the Receiver project. 
+ 1. In the Sender's console is the text `Press <enter> to send a message` when the app is ready.
+ 1. Hit `<enter>`.
+
+
+## Verifying that the sample works correctly
+
+ 1. The Receiver displays information that an order was submitted.
+ 1. The Sender displays information that the order was accepted.
+ 1. Hit `<enter>` in the Receiver console to shut it down.
+ 1. Go to the SQL Server Management Studio and delete the `receiver` database.
+ 1. Hit `<enter>` again in the Sender console
+ 1. Notice the retry mechanism kicks in after a 10 seconds delay. It attempts to retry sending the message to the destination database every 10 seconds.
+ 1. Create the `receiver` database again and restart the Receiver project.
+ 1. Notice that the message has been delivered to the Receiver.
+
+
+## Code walk-through
+
+When SQL Server transport is used in the [*multi-instance* mode](/nservicebus/sqlserver/deployment-options.md#modes-overview-multi-instance), the messages are directly inserted into the destination table in another database. If the receiving endpoint's database is down or inaccessible, e.g. because of network failures, the sending endpoint can't send messages to it. In such situation the exception is thrown from the `Send()` or the `Publish()` method, and the message can be lost.
+
+Those problems can be prevented by adding [store-and-forward functionality](/nservicebus/architecture/principles.md#drilling-down-into-details-store-and-forward-messaging) to the SQL Server transport, as explained in this sample. 
+
+NOTE: The [Outbox](/nservicebus/outbox/) would not help solve the issue presented in this example, because it is bypassed when sending messages from outside of a message handler e.g. from MVC controller.
+
+The sample contains three projects:
+
+ * Shared - A class library containing common code including messages definitions.
+ * Sender - A console application responsible for sending the initial `OrderSubmitted` message and processing the follow-up `OrderAccepted` message.
+ * Receiver - A console application responsible for processing the order message.
+
+Sender and Receiver use different databases, just like in a production scenario where two systems are integrated using NServiceBus. Each database contains, apart from business data, queues for the NServiceBus endpoint and tables for NServiceBus persistence.
+
+
+### Sender project
+
+The Sender does not store any data. It mimics the front-end system where orders are submitted by the users and passed via the bus to the back-end. It is configured to use SQL Server transport with NHibernate persistence.
+
+snippet:SenderConfiguration
+
+The Sender registers two custom behaviors, one for the send pipeline and the other for the receive pipeline.
+
+
+#### Send pipeline
+
+The new behavior is added at the beginning of the send pipeline (in Version 5) or in the routing stage (in Version 6).
```

Comment:
To avoid versioning hell I'd replace "Version 5/6" with "NServiceBus Version5/6".